corehq/apps/api/resources/v0_5.py

```diff
@@ -487,10 +488,22 @@ def get_resource_uri(self, bundle_or_obj=None, url_name='api_dispatch_detail'):
             obj = bundle_or_obj.obj
         else:
             obj = bundle_or_obj
-        return reverse('api_dispatch_detail', kwargs=dict(resource_name=self._meta.resource_name,
-                                                          domain=obj.domain,
-                                                          api_name=self._meta.api_name,
-                                                          pk=obj._id))
+        return self._get_resource_uri(obj)
+
+    def _get_resource_uri(self, obj):
+        # This function is called up to 1000 times per request
+        # so build url from a known string template
+        # to avoid calling the expensive `reverse` function each time
```

Comment:
`reverse` is expensive?  I had no idea.

Reply:
Yeah if you do it once it doesn't show up on the map, but I guess it turns out if you do it 1000 times, then it's notably slower than returning a predefined string 1000 times